### PR TITLE
data menu infrastructure

### PR DESCRIPTION
**Pull request by kecnry**
### Description

This pull request builds the infrastructure for the currrent viewer legend (which will eventually become the new data menu) to eventually be accessible via `viewer.data_menu` using the same framework as the sidebar plugins (with user-APIs and a direct connection between the python and vue template, rather than relying on the `viewer_item` defined in `app.py`).

### Change log entry

- [ ] Is a change log needed? If yes, is it added to `CHANGES.rst`? If you want to avoid merge conflicts,
  list the proposed change log here for review and add to `CHANGES.rst` before merge. If no, maintainer
  should add a `no-changelog-entry-needed` label.

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [ ] Are two approvals required? Branch protection rule does not check for the second approval. If a second approval is not necessary, please apply the `trivial` label.
- [ ] Do the proposed changes actually accomplish desired goals? Also manually run the affected example notebooks, if necessary.
- [ ] Do the proposed changes follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Are tests added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Are docs added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Did the CI pass? If not, are the failures related?
- [ ] Is a milestone set? Set this to bugfix milestone if this is a bug fix and needs to be released ASAP; otherwise, set this to the next major release milestone. Bugfix milestone also needs an accompanying backport label.
- [ ] After merge, any internal documentations need updating (e.g., JIRA, Innerspace)?

### CODE REVIEW

**Review by javerbukh — APPROVED** [review]
Looking forward to seeing where this goes! Couldn't break anything and the changes make sense, so I'll approve.

**Comment by pllim** [comment]
Will review next week. Thanks for your patience!

File: jdaviz/app.py [hunk 1]
```diff
@@ -2497,10 +2511,9 @@ def _create_viewer_item(self, viewer, vid=None, name=None, reference=None,
             'name': name or vid,
             'widget': "IPY_MODEL_" + viewer.figure_widget.model_id,
             'toolbar': "IPY_MODEL_" + viewer.toolbar.model_id if viewer.toolbar else '',  # noqa
-            'layer_options': "IPY_MODEL_" + viewer.layer_options.model_id,
-            'viewer_options': "IPY_MODEL_" + viewer.viewer_options.model_id,
+            'data_menu': 'IPY_MODEL_' + viewer._data_menu.model_id if hasattr(viewer, '_data_menu') else '',  # noqa
+            # TODO: remove unused entries after old data menu deprecation period
```
Comment (pllim) [inline jdaviz/app.py:2515]: What is the thing for "old data menu" that we can grep for to find this again when deprecation period is done?

**Review by pllim — COMMENTED** [review]
My understanding is that there are more things you need to add to this PR, so I only glanced at the code as-is for now. Code looks like code.

File: jdaviz/app.py (hunk 1 above)
Comment (kecnry) [reply jdaviz/app.py:2515]: I don't know that there is anything easily greppable, but I added links back to all these comments from the ticket [:cat:](https://jira.stsci.edu/browse/JDAT-4508)

**Comment by kecnry** [comment]
> My understanding is that there are more things you need to add to this PR

As far as I know, this is ready for review/merge.  I vote to defer building the UI for the menu itself until the ticket that populates with layers, since that may motivate the design a bit further.

**Review by pllim — APPROVED** [review]
Let's get it in then. Thanks!

**Pull request merged**